src/test/java/edu/harvard/iq/dataverse/api/DuplicateFilesIT.java

```diff
@@ -419,4 +420,77 @@ public void existingDirectoryPassLabelChangeDescription() throws IOException {
 
     }
 
+    /**
+     * This test is for the following scenario.
+     *
+     * What if the database has null for the directoryLabel? What if you pass in
+     * directory as “” (because you don’t realize you can just not pass it).
+     * when it check and compares, the old directory is null. so will that mean
+     * labelChange = true and it will fail even though you didn’t really change
+     * the directory?
+     *
+     * While "labelChange" does end up being true,
+     * IngestUtil.conflictsWithExistingFilenames returns false, so the change is
+     * allowed to go through. The description is allowed to be changed and the
+     * directoryLabel remains null even though the user passed in an empty
+     * string, which is what we want.
```

Comment:
The issue here is that if it gets to the conflictswithexisting stage and that returns false, I think we have a problem:

if conflictswithexisting filenames returns false, it’s because it’s treating ‘’ and null differently, since the name is the same). So if you did the same thing, but actually changed the name of the file to a different file name that does exist? Wouldn’t conflictswithexisting still return false then and allow you to rename to an already existing file.

The use case is you have two files:
• the existing values are null for directory for both, names are file1 and file2
• call the api to edit file1 with “” as directoryLabel and label as “file2"
• I’m concerned that labelChange would be true and conflictswithexisting would be false and that would allow you to have the same null/file2 for both.

Reply:
I'll add, one solution could be to not store null for directoryLabel ever. We could either store "" or even "/". This would also help in adding a unique constraint, since unique constraints are problematic with nulls.